src/utils/units.cpp

```diff
@@ -0,0 +1,115 @@
+//========================================================================================
+// (C) (or copyright) 2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "units.hpp"
+
+namespace ArtemisUtils {
+
+constexpr Real Msolar = 1.988416e33;
+constexpr Real AU = 1.495978707e13;
+constexpr Real Year = 31536000;
+constexpr Real parsec = 3.0857e18;
+constexpr Real Rjup = 6.991100e6;
+constexpr Real Mjup = 1.8982e30;
+
+Units::Units(ParameterInput *pin, std::shared_ptr<StateDescriptor> pkg) {
+  std::string unit_system_str = pin->GetOrAddString("artemis/units", "type", "scalefree");
+  if (unit_system_str == "scalefree") {
+    unit_system_ = UnitSystem::scalefree;
+  } else if (unit_system_str == "cgs") {
+    unit_system_ = UnitSystem::cgs;
+  } else {
+    PARTHENON_FAIL("Unit system not recognized! Valid choices are [scalefree, cgs]");
+  }
+
+  if (unit_system_ == UnitSystem::scalefree) {
+    length_ = 1.;
+    time_ = 1.;
+    mass_ = 1.;
+  } else {
+    std::string unit_specifier = pin->GetOrAddString("artemis", "specifier", "base");
+    if (unit_specifier == "base") {
+      length_ = pin->GetReal("artemis", "length");
+      time_ = pin->GetReal("artemis", "time");
+      mass_ = pin->GetReal("artemis", "mass");
+    } else if (unit_specifier == "ppd") {
+      length_ = AU;
+      mass_ = Msolar;
+      time_ = Year / (2. * M_PI);
+    } else {
```

Comment:
I don't understand the difference between `artemis/specifier` and `artemis/units/type`? I thought it would just be 

```
<artemis>
...
units = scalefree  # scalefree, cgs, ppd
length = ..
mass = ...
time = ...
```

Reply:
`[scalefree, cgs]` and `[base, ppd]` are separate ideas; the first fixes the physical units through setting the values of fundamental constants (either all `1` or cgs values) and `[base, ppd]` is different ways of setting the conversion between physical and code units, with `base` requiring the user to set length, mass, and time conversions, and `ppd` is a default set of those conversions for convenience with no user-settable parameters.

But I didnt completely remove the `units` node here so I will turn this into `artemis/physical_units` and `artemis/unit_conversion`